### PR TITLE
Generate and store federation OpenAPI spec in source tree

### PR DESCRIPTION
**Pull request by mbohlool**
A follow up of #33628 to also add federation spec to source tree.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34001)

### CODE REVIEW

File: federation/apis/openapi-spec/api.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
+{
+  "swagger": "2.0",
+  "info": {
+   "title": "Generic API Server /api",
```
Comment (nikhiljindal) [inline federation/apis/openapi-spec/api.json]: Need to fix this.

File: federation/apis/openapi-spec/logs.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  "paths": {
+   "/logs/": {
+    "get": {
+     "schemes": [
+      "https"
+     ],
+     "operationId": "logFileListHandler",
+     "responses": {
+      "default": {
+       "description": "Default Response."
```
Comment (nikhiljindal) [inline federation/apis/openapi-spec/logs.json]: What is this supposed to mean?

File: federation/apis/openapi-spec/v1beta1.extensions.json [hunk 3]
```diff
@@ -0,0 +1,104 @@
+{
+  "swagger": "2.0",
+  "info": {
+   "title": "Generic API Server /apis/extensions",
+   "version": "unversioned"
+  },
+  "paths": {
+   "/apis/extensions/": {
```
Comment (nikhiljindal) [inline federation/apis/openapi-spec/v1beta1.extensions.json]: What about `apis/extensions/v1beta1/replicasets`?

File: federation/apis/openapi-spec/v1beta1.federation.json [hunk 4]
```diff
@@ -0,0 +1,104 @@
+{
+  "swagger": "2.0",
+  "info": {
+   "title": "Generic API Server /apis/federation",
+   "version": "unversioned"
+  },
+  "paths": {
+   "/apis/federation/": {
```
Comment (nikhiljindal) [inline federation/apis/openapi-spec/v1beta1.federation.json]: same. what about `/apis/federation/v1/clusters`?

File: federation/apis/openapi-spec/logs.json (hunk 2 above)
Comment (mbohlool) [inline federation/apis/openapi-spec/logs.json]: If a route does not provide a response, a default response will be used (provided as config to the spec generator too). We can change this for federation or kubernetes, what do you think it should be?

File: federation/apis/openapi-spec/v1beta1.extensions.json (hunk 3 above)
Comment (mbohlool) [inline federation/apis/openapi-spec/v1beta1.extensions.json]: There were a bug in generation code fixed by #34150, I will regenerate these after the PR is merged.

File: federation/apis/openapi-spec/v1beta1.federation.json (hunk 4 above)
Comment (mbohlool) [inline federation/apis/openapi-spec/v1beta1.federation.json]: ditto

File: federation/apis/openapi-spec/logs.json (hunk 2 above)
Comment (nikhiljindal) [inline federation/apis/openapi-spec/logs.json]: Whats an example value?
We should probably print that config value here?

**Comment by nikhiljindal** [comment]
Cool should be good to merge once you have regenerated the spec and ensured that no resource is missing.

Next step will be to ensure this works on Macs and then add a verify script and add it to post commit.
Thanks

**Comment by mbohlool** [comment]
I've regenerated files with the fix. Those resources look to be there, please take another look.

**Comment by mbohlool** [comment]
BTW, It is working on mac now by removing sudo from the script. PTAL.

**Pull request merged**